packages/ruleset-migrator/src/transformers/formats.ts

```diff
@@ -2,32 +2,34 @@ import { builders as b, namedTypes } from 'ast-types';
 import { Transformer, TransformerCtx } from '../types';
 import { assertArray, assertString } from '../validation';
 
-const REPLACEMENTS = [
-  'oas2',
-  'oas3',
-  'oas3.0',
-  'oas3.1',
-  'asyncapi2',
-  'json-schema',
-  'json-schema-loose',
-  'json-schema-draft4',
-  'json-schema-draft6',
-  'json-schema-draft7',
-  'json-schema-2019-09',
-  'json-schema-2020-12',
-].reduce((replacements, id) => {
-  replacements[id] = id.replace(/\./g, '_').replace(/-([a-z])/g, (match, char) => String(char).toUpperCase());
-  return replacements;
-}, {});
+import schema from '../validation/schema';
+
+const ALIASES: Record<string, string> = {
+  'json-schema-2019-09': 'json-schema-draft-2019-09',
+  'json-schema-2020-12': 'json-schema-draft-2020-12',
+};
+
+const REPLACEMENTS = Object.fromEntries(
+  schema.properties.formats.items.enum.map(format => [
+    format,
+    (ALIASES[format] ?? format)
+      .replace(/\.|(?<=[0-9])-(?=[0-9])/g, '_')
+      .replace(/-([0-9a-z])/g, (match, char) => String(char).toUpperCase()),
+  ]),
+);
 
 function transform(ctx: TransformerCtx, input: unknown): namedTypes.ArrayExpression {
   assertArray(input);
 
   return b.arrayExpression(
-    input.map(format => {
-      assertString(format);
-      return ctx.tree.addImport(REPLACEMENTS[format], '@stoplight/spectral-formats');
-    }),
+    Array.from(
+      new Set(
```

Comment:
Although is not invalid for formats to be listed more than once, it feels nice to have them de-duped.